src/datautils/wellboreItemShapeGenerator.ts

```diff
@@ -25,30 +25,24 @@ export const makeTubularPolygon = (rightPath: Point[], leftPath: Point[]): Point
   ];
 };
 
-export const findCasing = (id: string, casings: Casing[]): Casing => {
-  const res = casings.filter((c) => c.casingId === id);
-  if (res.length === 0) {
-    throw new Error('Casing not found');
-  }
-  return res[0];
-};
-
 export const overlaps = (top1: number, bottom1: number, top2: number, bottom2: number): boolean => top1 <= bottom2 && top2 <= bottom1;
 
 export const uniq = <T>(arr: T[]): T[] => Array.from<T>(new Set(arr));
 
 export const findIntersectingItems = (
   cement: Cement,
-  parentCasing: Casing,
+  bottomOfCement: number,
+  parentCasings: Casing[],
   casings: Casing[],
   holes: HoleSize[],
 ): { holes: HoleSize[]; outerCasings: Casing[] } => {
   const { toc: start } = cement;
-  const { end } = parentCasing;
-  const overlappingHoles = holes.filter((h: HoleSize) => overlaps(start, end, h.start, h.end) && h.diameter > parentCasing.diameter);
-  const overlappingCasings = casings.filter(
-    (c: Casing) => c.casingId !== cement.casingId && overlaps(start, end, c.start, c.end) && c.diameter > parentCasing.diameter,
-  );
+  const end = bottomOfCement;
```

Comment:
Redundant variable? I reckon it is used only for clarity?!

Reply:
yes. The inputs are a bit confusing. We should maybe be reconsider the data model for cement and make bottom of cement required on it.